libs/coin-modules/coin-aptos/src/logic/combine.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  AccountAuthenticatorEd25519,
+  Deserializer,
+  Ed25519PublicKey,
+  Ed25519Signature,
+  Hex,
+  RawTransaction,
+  type SimpleTransaction,
+  generateSignedTransaction,
+} from "@aptos-labs/ts-sdk";
+
+export function combineSignedTransaction(
+  txRaw: RawTransaction,
+  signature: Ed25519Signature,
+  pubkey: Ed25519PublicKey,
+): Uint8Array {
+  const authenticator = new AccountAuthenticatorEd25519(pubkey, signature);
+
+  return generateSignedTransaction({
+    transaction: { rawTransaction: txRaw } as SimpleTransaction,
+    senderAuthenticator: authenticator,
+  });
+}
+
+export function combine(tx: string, signature: string, pubkey?: string) {
+  if (tx === undefined) {
+    throw new Error("tx must be provided");
+  }
+
+  if (!Hex.isValid(tx)) {
+    throw new Error("tx must be a valid hex value");
+  }
+
+  if (signature === undefined) {
```

Comment:
This check is not useful (mandatory input)

Reply:
removed, thanks.